src/main/webapp/scripts/view_card.js

```diff
@@ -13,42 +13,81 @@
 // limitations under the License.
 
 /**
- * Query salecard from servlet
- * TODO: fetch from servlet
+ * Query salecard from servlet, draw map and show vendor's information.
  */
-const querySalecard = () => {
-  let salecard = vendor.saleCard;
+async function querySalecard() {
+  await updateLocation();
+
+  const params = new URLSearchParams();
+  params.append('vendorId', getVendorId());
+  params.append('lat', document.getElementById('lat').value);
+  params.append('lng', document.getElementById('lng').value);
+
+  let vendor;
+  await fetch('/get-vendor?' + params)
+      .then(response => response.json())
+      .then(vendorEntity => {
+        vendor = vendorEntity;
+      });
```

Comment:
What if the request goes wrong and the servlet sends an error?

Reply:
In that case, what do you recommend ? Send an alert, redirect to the home page ... ?

Reply:
Hmm... yeah, displaying an alert before redirecting would be fine.